sklearn/utils/estimator_checks.py

```diff
@@ -495,15 +496,16 @@ def check_estimator(Estimator, generate_only=False):
             warnings.warn(str(exception), SkipTestWarning)
 
 
-def _boston_subset(n_samples=200):
-    global BOSTON
-    if BOSTON is None:
-        X, y = load_boston(return_X_y=True)
-        X, y = shuffle(X, y, random_state=0)
-        X, y = X[:n_samples], y[:n_samples]
+def _regression_dataset(n_samples=200):
```

Comment:
Not a blocker. The first call to `_regression_dataset` will create a dataset with `n_samples`, and for all proceeding calls `n_samples` will do nothing.

At this point, I would just define `REGRESSION_DATASET` on top 

```py
REGRESSION_DATASET = make_regression(...)
```

And then have the checks do:

```py
X, y = REGRESSION_DATASET
```

Reply:
True